src/main/java/net/mcreator/ui/dialogs/wysiwyg/SliderDialog.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2025, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.dialogs.wysiwyg;
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.parts.gui.GUIComponent;
+import net.mcreator.element.parts.gui.Slider;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.IHelpContext;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.procedure.ProcedureSelector;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.validators.JavaMemberNameValidator;
+import net.mcreator.ui.validation.validators.UniqueNameValidator;
+import net.mcreator.ui.wysiwyg.WYSIWYG;
+import net.mcreator.ui.wysiwyg.WYSIWYGEditor;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+
+public class SliderDialog extends AbstractWYSIWYGDialog<Slider> {
+
+	public SliderDialog(WYSIWYGEditor editor, @Nullable Slider slider) {
+		super(editor, slider);
+		setModal(true);
+		setSize(460, 390);
+		setLocationRelativeTo(editor.mcreator);
+		setTitle(L10N.t("dialog.gui.slider_add_title"));
+
+		VTextField sliderMachineName = new VTextField(15);
+		UniqueNameValidator validator = new UniqueNameValidator(L10N.t("dialog.gui.slider_name_validator"),
+				sliderMachineName::getText, () -> editor.getComponentList().stream().map(GUIComponent::getName),
+				new JavaMemberNameValidator(sliderMachineName, false));
+		validator.setIsPresentOnList(slider != null);
+		sliderMachineName.setValidator(validator);
+		sliderMachineName.enableRealtimeValidation();
+
+		JTextField sliderPrefix = new JTextField(8);
+		JSpinner minValueSpinner = new JSpinner(new SpinnerNumberModel(0, -10000.0, 10000.0, 1));

```

Comment:
For min/max selection, use JMinMaxSpinner

I would also allow values from min to max float, not just to +/- 10000, unless there is some limit in the game

Reply:
I'll change it to JMinMaxSpinner (didn't know it exist, my bad).

However, about allowing float values, users can already write any number between -10k and +10k, as shown with the .0 at the end of each. I put the step value to 1 as I thought it would be the most common value used by users, but everything else in the code is made to support a double value.

Reply:
I can confirm that using decimal values inside the dialog works fine and cause no problem at all in the software or in the game. Mojang supports double values (and so NeoForge's `ExtendedSlider` class too) and this works perfectly with the current design.

Reply:
I did not mean that you should not use decimal values. I just meant that instead of hardcoded 10000 min max we can allow min and max float value instead

Reply:
Oh ok, I didn't understand that. Understood

Reply:
Yeah, I'll use the Double.MAX and MIM static variables. I didn't understand you were talking of them.

Reply:
Done